test/api/README.md

```diff
@@ -0,0 +1,21 @@
+# Testing the T-Route FastAPI Extension
+
+The following folder contains data files that are to be used to test the T-Route FastAPI code within src/app
+
+To use these files, follow the steps below:
+
+1. Copy the `test_compose.yaml` file in the root project dir 
+2. Run `docker compose -f test_compose.yaml up`
+3. visit `localhost:8004/docs` in your browser
+4. Enter the following parameters into the `/api/v1/flow_routing/v4` endpoint
+- lid=CAGM7
+- feature_id=2930769
+- hy_id=1074884
+- initial_start=0
+- start_time=2024-08-24T00:00:00
+- num_forecast_days=5
+5. Click execute
+6. A Status 200 code means the run ran, and test/api/data/troute_output will be populated in the `{lid}/` folder
+
```

Comment:
I am getting below error
![image](https://github.com/user-attachments/assets/0fd888a1-00ba-488f-b50c-88c9a3e4e5d8)

Reply:
Can you post a screenshot of the the terminal that has docker compose running (where you ran `docker compose -f test_compose.yaml up`)?

I just ran through the test case and got the following output, which matches what T-Route would show if I were to be running the code locally:

My GET request looks like:
```shell
troute-1  | INFO:     172.22.0.1:54084 - "GET /api/v1/flow_routing/v4/?lid=CAGM7&feature_id=2930769&hy_id=1074884&initial_start=0&start_time=2024-08-24T00%3A00%3A00&num_forecast_days=5 HTTP/1.1" 200 OK
```
<img width="2023" alt="image" src="https://github.com/user-attachments/assets/0ffed93a-6043-439a-b444-8eaa7838f865">

Reply:
The issue was that the `test_compose.yaml` file was in the wrong directory, but it's working now.

A couple more questions/comments:

When I run it, a browser opens to `http://localhost:8000/`, but I have to manually change it to `http://localhost:8004/docs` to display the correct page. Is there a way to open the correct page directly?

Could you provide an example of how to run the Lower Colorado test using `//t-route/test/LowerColorado_TX_v4/test_AnA_V4_HYFeature.yaml`?

Note: Previously typed in wrong place

Reply:
> When I run it, a browser opens to http://localhost:8000/, but I have to manually change it to http://localhost:8004/docs to display the correct page. Is there a way to open the correct page directly?

I changed the `test_compose.yaml` file to point to port 8000. The reason it was on port 8004, rather than 8000 to begin with, was because of testing with the Replace and Route application. Since this compose file is local to T-Route, changing the FastAPI port back to 8000 will have no effect on RnR.

Reply:
As Amin mentioned, testing your application with //t-route/test/LowerColorado_TX_v4/test_AnA_V4_HYFeature.yaml would be more relevant for us to review your PR, as it will help determine whether the T-Route Docker image produces the same output as the current T-Route version and behaves as expected in terms of compute speed and stability.

Reply:
Thanks, @kumdonoaa and @AminTorabi-NOAA for the suggestion. I just pushed updates to the `test/api/README.md` file for how to run this API with the `LowerColorado` data. 

Some things to note:
1. I copied the `LowerColorado` channel forcings and v20.1 hydrofabric gpkg to the same directories that the `test/api/data` is located. This was to prevent having to change the mounted volume location within the `test_compose.yaml` file
2. You'll notice in the instructions I added (see below) the `feature_id` and `hy_id` are the same. I put this this way because of how RnR calls this API. RnR saves it's geopackages based on their `feature_id of their downstream point`, and thus we pass in the feature ID to t-route so it knows where to look for the gpkg data. In the `LowerColorado` case, we just want to make sure we know where the data. 
```md
- lid=LowerColorado
- feature_id=2420801
- hy_id=2420801
- initial_start=0
- start_time=2023-04-01T00:00:00
- num_forecast_days=2
```
3. The num_forecast days is set to 2 since the data is from 4/01/23-4/03/23